src/connectors/query-rules/connectQueryRules.ts

```diff
@@ -87,7 +90,9 @@ function getRuleContextsFromTrackedFilters({
   const ruleContexts = Object.keys(trackedFilters).reduce<string[]>(
     (facets, facetName) => {
       const facetRefinements: TrackedFilterRefinement[] = getRefinements(
-        helper.lastResults || {},
+        // Note that this isn't correct, but getRefinements only uses properties,
```

Comment:
We should make it explicit that this comment is about the typing.

```suggestion
        // An empty object is technically not a `SearchResults` but `getRefinements`
        // only accesses properties, meaning it will not throw with an empty object.
```

Can't we work around that by changing the `getRefinements` signature with only partial search parameters?